### DISCUSSION TITLE
Unbelievably slow

### DISCUSSION THREAD

**Discussion by step135** [pockethost/pockethost#357] [Q&A]
I created one document of two fields for all database and it takes each time 3 seconds to retrieve it. What's wrong!?

**Comment by benallfree** [comment]
@step135 I'll need a lot more info to troubleshoot this. Please join us on Discord https://discord.com/channels/1128192380500193370/1128192380500193373 if you are able, your questions will be answered faster.

Otherwise, please post an HTTP GET example here.

**Comment by step135** [comment]
I found on Reddit other people complaining about the same and having the same performance. It looks like you serve the data from far away Europe. But still it doesn't explain those 3 seconds (consequence of OPTION and GET requests, each of them 1.5 seconds).

**Reply by benallfree** [reply]
Could you provide the Reddit link please?

Also, please provide the HTTP GET in question.

Latency is complex and is discussed often on Discord. Just today, someone was reporting 1s latency while someone else in the same country was getting 200ms just like I was.

**Reply by step135** [reply]
`GET https://{xxx}.pockethost.io/api/collections/posts/records?page=1&perPage=500&skipTotal=1&sort=-created`

1s latency is a lie, it can never be achieved with any service !!!

**Reply by benallfree** [reply]
Nobody can help you until you will provide a valid URL.

**Comment by lukepighetti** [comment]
Hey there, I know this isn't enough info to do a full repro but I am seeing some pretty alarming latencies and 429s on an app that is trying to do batch inserts. When the insert returns 429 the admin panel stops responding for about 30 seconds. In my case I'm checking if a record exists (with a unique field) before creating a record(5 columns, < 20kb in size). I'm doing this about 100 times in 30 seconds and getting throttled.

**Reply by lukepighetti** [reply]
the interesting thing is it's not a consistent issue. having no issues right now but was very troublesome a few moments ago.

**Reply by lukepighetti** [reply]
I was also seeing 15 seconds or more latency on authentication requests

**Reply by benallfree** [reply]
429 means you are hitting a rate limiting error which is currently 50 requests per 10 seconds per IP. Please move over to https://discord.gg/nVTxCMEcGT for better support :)

**Reply by lukepighetti** [reply]
appreciate the info and thank you for the discord link. What's the recommended way to do bulk inserts remotely?

**Reply by benallfree** [reply]
https://www.npmjs.com/package/bottleneck 💪

**Reply by lukepighetti** [reply]
awesome... thank you!
